modules/network/src/main/java/org/apache/ignite/internal/network/ssl/KeystoreLoader.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.network.ssl;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import org.apache.ignite.internal.network.configuration.KeyStoreView;
+
+/** Java keystore loader. */
+public final class KeystoreLoader {
+
+    private KeystoreLoader() {
+        throw new IllegalStateException("KeystoreLoader is an utility class and should not be instantiated");
+    }
+
+    /** Initialize and load keystore with provided configuration. */
+    public static KeyStore load(KeyStoreView keyStoreConfiguration)
+            throws KeyStoreException, IOException, CertificateException, NoSuchAlgorithmException {
```

Comment:
Do we really need to throw all these exceptions? Maybe we should wrap them into `IgniteException` here?

Reply:
They are caught by the calling side that decides how to interpret it and wraps to IgniteException with the appropriate error code.

Reply:
Are they handled differently? Looks like I missed that

Reply:
I would say no, they are handled in the same way. 

But `KeystoreLoader` is used in `SslContextProvider` together with `TrustManagerFactory` and `KeyManagerFactory`. These classes declare almost the same set of checked exceptions in their signatures. This fact forces us to catch them in `SslContextProvider `. So, there is a place, that handles all exceptions:

```java 
        } catch (NoSuchFileException e) {
            throw new IgniteException(Common.SSL_CONFIGURATION_ERR, String.format("File %s not found", e.getMessage()), e);
        } catch (IOException | CertificateException | KeyStoreException | NoSuchAlgorithmException | UnrecoverableKeyException e) {
            throw new IgniteException(Common.SSL_CONFIGURATION_ERR, e);
        }

Reply:
ok, looks fine either way